pkg/operator/etcdcertsigner/target_cert_creator.go

```diff
@@ -0,0 +1,70 @@
+package etcdcertsigner
+
+import (
+	"bytes"
+	"crypto/x509"
+	"fmt"
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/library-go/pkg/operator/certrotation"
+	corev1 "k8s.io/api/core/v1"
+	"time"
+)
+
+// CARotatingTargetCertCreator ensures we also rotate leaf certificates when we detect a change in signer.
+// The certrotation.TargetCertCreator only assumes the bundle to change on a CA rotation, whereas we have to keep
+// the bundle around for some time for a proper static pod rollout.
+type CARotatingTargetCertCreator struct {
+	certrotation.TargetCertCreator
+}
+
+func (c *CARotatingTargetCertCreator) NeedNewTargetCertKeyPair(
```

Comment:
So I get that we're overriding the `NeedNewTargetCertKeyPair()` method here to check the extra constraints on the signer not matching to trigger rotation.

Shouldn't we be using `CARotatingTargetCertCreator` somewhere then?
e.g:
https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/tlshelpers/tlshelpers.go#L216
or
https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/tlshelpers/tlshelpers.go#L109

(Unless I'm being dense and missing where we have that already).

Reply:
LOL you're right. Sorry, I'll get that done throughout the day.

Reply:
done